Example/Tests/DSProviderTransactionsTests.m

```diff
@@ -179,15 +179,11 @@ - (void)testCollateralProviderRegistrationTransaction {
 
     XCTAssertEqual(providerRegistrationTransaction.port, providerRegistrationTransactionFromMessage.port, @"Provider transaction port doesn't match up");
 
-    XCTAssertEqualObjects(providerRegistrationTransaction.inputHashes, providerRegistrationTransactionFromMessage.inputHashes, @"Provider transaction input hashes are having an issue");
+    XCTAssertEqualObjects(providerRegistrationTransaction.inputs, providerRegistrationTransactionFromMessage.inputs, @"Provider transaction inputs are having an issue");
```

Comment:
I think you will need to implement an equality operation for this to work on DSTransactionInput. (Same for Output)